src/sqlfluff/core/parser/segments/base.py

```diff
@@ -1008,13 +1048,22 @@ def apply_fixes(self, dialect, rule_code, fixes):
                 else:
                     seg = todo_buffer.pop(0)
 
-                    fix_buff = fixes.copy()
-                    unused_fixes = []
-                    while fix_buff:
-                        f = fix_buff.pop()
-                        # Look for identity not just equality.
-                        # This handles potential positioning ambiguity.
-                        if f.anchor is seg:
```

Comment:
Below, I reworked the core `apply_fixes()` logic:
* Consumes a dictionary of `AnchorEditInfo` rather than a list of fixes
* Add the ability to handle `create_before` and `create_after` the same anchor (new requirement for L053)